From ioe-lkml@rameria.de  Thu Jul 31 09:58:18 2008
From: Ingo Oeser <ioe-lkml@rameria.de>
Date: Wed, 23 Jul 2008 01:25:01 +0200
Subject: kobject: Replace ALL occurrences of '/' with '!' instead of only the first one.
To: Greg KH <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org, Kay Sievers <kay.sievers@vrfy.org>
Message-ID: <200807230125.02532.ioe-lkml@rameria.de>
Content-Disposition: inline


From: Ingo Oeser <ioe-lkml@rameria.de>

A recent patch from Kay Sievers <kay.sievers@vrfy.org>
replaced the first occurrence of '/' with '!' as needed for block devices.

Now do some cheap defensive coding and replace all of them to avoid future
issues in this area.

Signed-off-by: Ingo Oeser <ioe-lkml@rameria.de>
Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/lib/kobject.c
+++ b/lib/kobject.c
@@ -223,8 +223,7 @@ static int kobject_set_name_vargs(struct
 		return -ENOMEM;
 
 	/* ewww... some of these buggers have '/' in the name ... */
-	s = strchr(kobj->name, '/');
-	if (s)
+	while ((s = strchr(kobj->name, '/')))
 		s[0] = '!';
 
 	kfree(old_name);
